### PR TITLE
Propagate DECIDE messages from the previous instance

### LINKED ISSUE #579: Avoid dropping decides from the previous instance
We currently drop all messages from the previous instance, but we should consider allowing decides through. Otherwise, some nodes may not be able to make progress until they can use the certificate exchange.

### PR DESCRIPTION
**Pull request by Stebalien**
fixes #579

### CODE REVIEW

**Comment by Stebalien** [comment]
The fuzzer now appears to be hanging pretty consistently. Not sure what's up with that but I'll try to debug it a bit.

**Comment by masih** [comment]
> The fuzzer now appears to be hanging pretty consistently.

I suspect it's the runners. I'll switch to IPDX runners with extended fuzz time to investigate further.

File: gpbft/participant.go [hunk 1]
```diff
@@ -542,6 +552,16 @@ func (p *Participant) Describe() string {
 	return p.gpbft.Describe()
 }
 
+func (p *Participant) tracing() bool {
+	return p.tracer != nil
+}
+
+func (p *Participant) trace(format string, args ...any) {
+	if p.tracer != nil {
```
Comment (masih) [inline gpbft/participant.go:560]:
```suggestion
	if p.tracing() {
```

File: gpbft/participant.go [hunk 2]
```diff
@@ -542,6 +552,16 @@ func (p *Participant) Describe() string {
 	return p.gpbft.Describe()
 }
 
+func (p *Participant) tracing() bool {
```
Comment (masih) [inline gpbft/participant.go:555]: Subjective: `tracingEnabled`?

**Review by masih — COMMENTED** [review]
IIUC, the sole purpose of this PR (and #579) is to **continue to propagate DECIDEs via pubsub**, because invalid messages will not be propagated at _pubsub layer_.  These changes make no difference to the operation of the core GPBFT protocol, because on `RecieveMessage` [all messages from previous instances are dropped](https://github.com/filecoin-project/go-f3/blob/50dd5c1801c0e575e53c1407ec825cdc0e6c022d/gpbft/participant.go#L386-L390) at _GPPBFT layer_.

Is my understanding correct @Stebalien? asking since #579 does't explicitly mention the scope of change is for pubsub layer only.

If so, this LGTM 👍

**Review by Kubuxu — APPROVED** [review]

**Comment by Stebalien** [comment]
> These changes make no difference to the operation of the core GPBFT protocol, because on RecieveMessage [all messages from previous instances are dropped](https://github.com/filecoin-project/go-f3/blob/50dd5c1801c0e575e53c1407ec825cdc0e6c022d/gpbft/participant.go#L386-L390) at GPPBFT layer

Yes.

**Review by masih — APPROVED** [review]

**Comment by Stebalien** [comment]
Fuzzing is failing on 'main' as well.

**Comment by Stebalien** [comment]
At least on my machine...

**Pull request merged**